presto-main/src/main/java/io/prestosql/sql/planner/ExpressionInterpreter.java

```diff
@@ -252,10 +251,10 @@ public Object evaluate()
         return visitor.process(expression, new NoPagePositionContext());
     }
 
-    public Object evaluate(int position, Page page)
+    public Object evaluate(SymbolResolver inputs)
```

Comment:
What's the difference between this `evaluate` and the `optimize` now? If you provide `SymbolResolver`, using `optimize` would be just fine, right?